doc/migrations/jid-from-mam-muc-script.md

```diff
@@ -0,0 +1,82 @@
+## The purpose of sender-jid-from-mam-message.escript
+
+This script may be used as a part of migration from MongooseIM 3.3.0 (or older).
+It is able to extract a groupchat message's sender's JID from an XML payload.
+This piece of information is essential for GDPR commands (retrieve data and remove user) to work properly, as without it the operations on MAM MUC data in DB would be extremely inefficient.
+
+Please consult "3.3.0 to..." migration guide for details.
+DB-specific sections describe where the payloads are stored and what you should do with the extracted JID.
+
+## Requirements
+
+This script may be executed in every \*nix environment which has OTP 19.0 (or newer) installed and `escript` executable is in `PATH`.
+
+It doesn't depend on any MongooseIM code or library, so it may be used as a standalone file.
+
+## How to use?
+
+`sender-jid-from-mam-message.escript (eterm | xml)`
+
+The only parameter required by the script is the input format.
+
+You should use `eterm` if (in MongooseIM config file):
+
+* You haven't set `db_message_format` option for MAM at all.
+* `db_message_format` is set to `mam_message_compressed_eterm` or `mam_message_eterm`
+
+You should use `xml` option if:
+* `db_message_format` is set to `mam_message_xml`.
+
+Once started, script will run in an infinite loop (until killed or interrupted), expecting a stream of inputs.
```

Comment:
Once started, the script will run in an infinite loop (until killed or interrupted), expecting a stream of inputs.